mergify_engine/actions/merge.py

```diff
@@ -41,14 +41,14 @@ class MergeAction(merge_base.MergeBaseAction):
         voluptuous.Required("rebase_fallback", default="merge"): voluptuous.Any(
             "merge", "squash", None
         ),
-        voluptuous.Required("strict", default=False): voluptuous.Any(
-            bool,
-            voluptuous.All("smart", voluptuous.Coerce(lambda _: "smart+ordered")),
-            voluptuous.All(
-                "smart+fastpath", voluptuous.Coerce(lambda _: "smart+fasttrack")
+        voluptuous.Required(
+            "strict", default=merge_base.StrictMergeParameter.false
```

Comment:
here's the catch: the default value will by passed to all your validators after.
Which mean the enum `false` will get through L46-L51.
However, it's not going to match the first one L48.

Your validators are correct though, so you need to pass a default value that works fine through all validator like a user provided value would 😄